### PR TITLE
implement o

### LINKED ISSUE #7: Parameter -o

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #7

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -71,4 +72,11 @@ func main() {
 	}
 	dataJoin := bytes.Join(dataSplit, []byte("\n"))
 	fmt.Printf("%s\n", strings.TrimPrefix(string(dataJoin), "\n"))
+
+	if *writingParameter != "" {
```
Comment (Al2Klimov) [inline main.go:75]: Your behavior differs from the original sort.

File: main.go [hunk 2]
```diff
@@ -71,4 +72,11 @@ func main() {
...
+		writeErr := ioutil.WriteFile(*writingParameter, dataJoin, 0644)
```
Comment (Al2Klimov) [inline main.go]: Your output differs from the output w/o -o.

File: main.go [hunk 3]
```diff
@@ -71,4 +72,11 @@ func main() {
...
+		if writeErr != nil {
+			fmt.Printf("Error occured while writing file %#v", writeErr)
```
Comment (Al2Klimov) [inline main.go]: This error reporting differs from the above ones.

File: main.go [hunk 4]
```diff
@@ -12,6 +12,7 @@ import (
 
 func main() {
 	numericParameter := flag.Bool("n", false, "Sort numbers from lowest to highest")
+	writingParameter := flag.String("o","","Save result into a file")
```
Comment (Al2Klimov) [inline main.go]: This formatting differs a bit from the other one.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go (hunk 4 above)
Comment (justinmueller05082000) [reply main.go]: i dont exactly get your point

Comment (Al2Klimov) [reply main.go]: You forgot the code formatting.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin feature/parameter-o-7`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol that compares sort vs your sort on -o w/ both inputs w/ tailing \n and inputs w/o tailing \n.

**Comment by justinmueller05082000** [comment]
**Test protocol:**

**GO CODE:**
```
package main

import (
	"bytes"
	"flag"
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
)

func main() {
	numericParameter := flag.Bool("n", false, "Sort numbers from lowest to highest")
	writingParameter := flag.String("o", "", "Save result into a file")
	flag.Parse()

	var data []byte

	if len(flag.Args()) < 1 {
		content, err := ioutil.ReadAll(os.Stdin)
		if err != nil {
			fmt.Println(err)
			return
		}
		data = content
	} else {
		fileContent, readFileErr := ioutil.ReadFile(flag.Args()[0])
		if readFileErr != nil {
			fmt.Println(readFileErr)
			return
		}
		data = fileContent
	}

	if string(data) == "" {
		return
	}

	dataSplit := bytes.Split(data, []byte("\n"))
	lengthOfData := len(dataSplit)

	for i := 0; i < lengthOfData; i++ {
		j := i
		for j > 0 {
			if !*numericParameter {
				if bytes.Compare(dataSplit[j], dataSplit[j-1]) < 0 {
					dataSplit[j], dataSplit[j-1] = dataSplit[j-1], dataSplit[j]
					j -= 1
				} else {
					break
				}
			} else {
				comparer1, convertErr1 := strconv.Atoi(string(dataSplit[j]))
				if convertErr1 != nil {
					fmt.Println(convertErr1)
					return
				}
				comparer2, convertErr2 := strconv.Atoi(string(dataSplit[j-1]))
				if convertErr2 != nil {
					fmt.Println(convertErr2)
					return
				}
				if comparer1 < comparer2 {
					dataSplit[j], dataSplit[j-1] = dataSplit[j-1], dataSplit[j]
					j -= 1
				} else {
					break
				}
			}
		}
	}
	dataJoin := bytes.Join(dataSplit, []byte("\n"))

	if *writingParameter != "" {
		writeErr := ioutil.WriteFile(*writingParameter, []byte(strings.TrimPrefix(string(dataJoin), "\n")+"\n"), 0644)
		if writeErr != nil {
			fmt.Println(writeErr)
		}
	} else {
		fmt.Printf("%s\n", strings.TrimPrefix(string(dataJoin), "\n"))
	}
}
```

**________________________________________**

**Case (1):**
**toSort.txt:**
```
11
23
54
4
8
436
7
2

```
**Terminal:**` go run . -o output.txt toSort.txt `
**output.txt:**
```
11
2
23
4
436
54
7
8

```

**Terminal:** `sort -o output.txt toSort.txt `
**output.txt:**
```
11
2
23
4
436
54
7
8

```

**Conclusion(1):**
Both results are equal.

**________________________________________**

**Case(2):**
**toSort.txt:**
```
11
23
54
4
8
436
7
2
```
**Terminal:**` go run . -o output.txt toSort.txt `
**output.txt:**
```
11
2
23
4
436
54
7
8

```

**Terminal:** `sort -o output.txt toSort.txt `
**output.txt:**
```
11
2
23
4
436
54
7
8

```

**Conclusion(2):**
Both results are equal.

**________________________________________**

**Final Conclusion:**
It doesn't matter if the last line of our toSort.txt is empty or not, the result will be the same in both cases. If there's no final empty line it will be created. If there's a final empty line the program does not change the order or formatting and just let it be as it is.

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**